src/tests/ffi-key-sig.cpp

```diff
@@ -468,18 +474,49 @@ TEST_F(rnp_tests, test_ffi_export_revocation)
     assert_rnp_success(
       rnp_ffi_set_pass_provider(ffi, ffi_string_password_provider, (void *) "wrong"));
     assert_rnp_failure(rnp_key_export_revocation(
-      key_handle, output, 0, "SHA256", "superseded", "test key revocation"));
+      key_handle, output, flags, "SHA256", "superseded", "test key revocation"));
+
+    rnp_output_destroy(output);
+    assert_rnp_success(rnp_key_handle_destroy(key_handle));
+    assert_rnp_success(rnp_ffi_destroy(ffi));
+}
+
+TEST_F(rnp_tests, test_ffi_export_revocation_failure_modes)
+{
+    for (bool armor: {false, true}) {
+        test_ffi_export_revocation_failure_modes_fn(armor);
+    }
+}
+
+static void
+test_ffi_export_revocation_fn(bool armor)
+{
+    int       flags = armor ? RNP_KEY_EXPORT_ARMORED : 0;
+    rnp_ffi_t ffi = NULL;
+    assert_rnp_success(rnp_ffi_create(&ffi, "GPG", "GPG"));
+    assert_true(import_sec_keys(ffi, "data/test_key_validity/alice-sec.asc"));
+
+    rnp_key_handle_t key_handle = NULL;
+    assert_rnp_success(rnp_locate_key(ffi, "userid", "Alice <alice@rnp>", &key_handle));
+    rnp_output_t output = NULL;
+    assert_rnp_success(rnp_output_to_path(&output, "alice-revocation.pgp"));
+
     /* unlocked key - must succeed */
     assert_rnp_success(rnp_key_unlock(key_handle, "password"));
-    assert_rnp_success(rnp_key_export_revocation(key_handle, output, 0, "SHA256", NULL, NULL));
-    assert_rnp_success(rnp_output_destroy(output));
+    assert_rnp_success(
+      rnp_key_export_revocation(key_handle, output, flags, "SHA256", NULL, NULL));
```

Comment:
> Added line #L507 was not covered by tests

I can't see how this warning and other by codecov could be right.
I can imagine a cause for it though: there are two calls, `rnp_key_export_revocation()` and `assert_rnp_success()`, and as I recall from debugging practice the compiler attributes both of them to the first line, 506. So Codecov is like "hey, i've never encountered running the line 507 according to my traces!".
We could check this guess by temporarily violating the coding style on purpose and refactoring the code to be in single long lines.
Does the above make sense to you?

Reply:
In this case we may ignore Codecov's warning and stick to the linting rules.